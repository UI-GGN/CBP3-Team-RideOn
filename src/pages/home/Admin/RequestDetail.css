.cardContainer {
  width: 90%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-right: 20px;
  margin-top: 40px;
}

.requestDetailCard {
  width: 350px;
  margin-right: 40px;
}

.cardContent {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.requestDetailPaper {
  width: 90%;
  margin-top: 40px;
  margin-bottom: 30px;
}

.requestDetailTableCell {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
}

.cellHeaderContent {
  margin-top: 10px;
  font-weight: bolder;
  font-size: 17px;
}

.cellContent {
  margin-top: 10px;
  font-size: 17px;
}
