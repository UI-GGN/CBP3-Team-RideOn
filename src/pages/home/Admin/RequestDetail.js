import * as React from "react";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import {Chip} from "@mui/material";
import Table from "@mui/material/Table";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableRow from "@mui/material/TableRow";
import Paper from "@mui/material/Paper";
import "./RequestDetail.css";
import {useLocation} from "react-router-dom";

function RequestDetail() {
  const location = useLocation();
  const rowData = location.state?.rowData;

  const getStatusColor = (status) => {
    if (status === "APPROVED") {
      return "primary";
    } else if (status === "PENDING") {
      return "warning";
    } else if (status === "REJECTED") {
      return "error";
    } else {
      return "default";
    }
  };

  return (
    <>
      <div className="cardContainer">
        <Card className="requestDetailCard">
          <CardContent>
            <div className="cardContent">
              <Typography
                className="heading"
                fontSize="1rem"
                fontWeight="bold"
                font="Roboto"
                color="text.primary"
                gutterBottom
              >
                Status
              </Typography>
              <Chip
                label={rowData.status}
                color={getStatusColor(rowData.status)}
                size="small"
              />
            </div>
          </CardContent>
        </Card>
        <Card className="requestDetailCard">
          <CardContent>
            <div className="cardContent">
              <Typography
                fontSize="1rem"
                fontWeight="bold"
                font="Roboto"
                color="text.primary"
                gutterBottom
              >
                Created At
              </Typography>
              <Typography fontSize="0.8rem">{rowData.createdAt}</Typography>
            </div>
          </CardContent>
        </Card>
        <Card sx={{width: "420px"}}>
          <CardContent>
            <div className="cardContent">
              <Typography
                fontSize="1rem"
                fontWeight="bold"
                font="Roboto"
                color="text.primary"
                gutterBottom
              >
                Modified At
              </Typography>
              <Typography fontSize="0.8rem">{rowData.updatedAt}</Typography>
            </div>
          </CardContent>
        </Card>
      </div>
      <Paper className="requestDetailPaper">
        <TableContainer sx={{maxHeight: 440}}>
          <Table aria-label="sticky table">
            <TableRow>
              <TableCell className="requestDetailTableCell">
                <div className="cellHeaderContent">Employee Name</div>
                <div className="cellHeaderContent">Email</div>
                <div className="cellHeaderContent">Project Code</div>
              </TableCell>
              <TableCell align="left">
                <div className="cellContent">
                  {rowData.raisedBy && rowData.raisedBy.name}
                </div>
                <div className="cellContent">
                  {rowData.raisedBy && rowData.raisedBy.email}
                </div>
                <div className="cellContent">{rowData.projectCode}</div>
              </TableCell>
            </TableRow>
            <TableRow>
              <TableCell className="requestDetailTableCell">
                <div className="cellHeaderContent">Pickup Location</div>
                <div className="cellHeaderContent">Drop Location</div>
                <div className="cellHeaderContent">Pickup Time</div>
              </TableCell>
              <TableCell align="left">
                <div className="cellContent">{rowData.pickupLocation}</div>
                <div className="cellContent">{rowData.dropLocation}</div>
                <div className="cellContent">{rowData.pickupTime}</div>
              </TableCell>
            </TableRow>
            {rowData && rowData.allotedVendor && (
              <TableRow>
                <TableCell className="requestDetailTableCell">
                  <div className="cellHeaderContent">Vendor Name</div>
                  <div className="cellHeaderContent">Vendor Contact</div>
                </TableCell>
                <TableCell align="left">
                  <div className="cellContent">
                    {rowData && rowData.allotedVendor && rowData.allotedVendor.name}
                  </div>
                  <div className="cellContent">
                    {rowData &&
                      rowData.allotedVendor &&
                      rowData.allotedVendor.contactNumber}
                  </div>
                </TableCell>
              </TableRow>
            )}
          </Table>
        </TableContainer>
      </Paper>
    </>
  );
}

export default RequestDetail;
