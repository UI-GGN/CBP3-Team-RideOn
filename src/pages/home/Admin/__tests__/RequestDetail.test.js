import {render} from "@testing-library/react";
import {BrowserRouter, useLocation} from "react-router-dom";
import RequestDetail from "../RequestDetail";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: jest.fn(),
}));
describe("Request Detail test", () => {
  beforeEach(() => {
    const mockLocation = {
      pathname: "/home/request/:id",
      state: {
        rowData: {
          status: "Approved",
          createdAt: "2023-08-08T16:39:28.242Z",
          updatedAt: "2023-08-08T16:39:28.242Z",
          raisedBy: {
            email: "test@gmail.com",
            name: "Anu",
          },
          projectCode: "Test1",
          pickupTime: "Tue 18 Jul 2023 9:30 PM",
          dropLocation: "Location-A",
          pickupLocation: "Location-B",
          allotedVendor: {
            contactNumber: "919005695229",
            name: "A1 Cab Services",
          },
        },
      },
    };
    useLocation.mockReturnValue(mockLocation);
  });
  it("renders correctly", () => {
    const {baseElement} = render(<RequestDetail />, {wrapper: BrowserRouter});
    expect(baseElement).toMatchSnapshot();
  });
});
