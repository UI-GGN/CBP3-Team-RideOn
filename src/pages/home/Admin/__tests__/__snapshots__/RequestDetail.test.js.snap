// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Detail test renders correctly 1`] = `
<body>
  <div>
    <div
      class="cardContainer"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root requestDetailCard css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="cardContent"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom heading css-84mn5j-MuiTypography-root"
              font="Roboto"
            >
              Status
            </p>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1fj5pxn-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
              >
                Approved
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root requestDetailCard css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="cardContent"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-84mn5j-MuiTypography-root"
              font="Roboto"
            >
              Created At
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-r9rcpu-MuiTypography-root"
            >
              2023-08-08T16:39:28.242Z
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-jhsodw-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="cardContent"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-84mn5j-MuiTypography-root"
              font="Roboto"
            >
              Modified At
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-r9rcpu-MuiTypography-root"
            >
              2023-08-08T16:39:28.242Z
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 requestDetailPaper css-1ps6pg7-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-41abqd-MuiTableContainer-root"
      >
        <table
          aria-label="sticky table"
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-sizeMedium requestDetailTableCell css-bsph2u-MuiTableCell-root"
            >
              <div
                class="cellHeaderContent"
              >
                Employee Name
              </div>
              <div
                class="cellHeaderContent"
              >
                Email
              </div>
              <div
                class="cellHeaderContent"
              >
                Project Code
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
            >
              <div
                class="cellContent"
              >
                Anu
              </div>
              <div
                class="cellContent"
              >
                test@gmail.com
              </div>
              <div
                class="cellContent"
              >
                Test1
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-sizeMedium requestDetailTableCell css-bsph2u-MuiTableCell-root"
            >
              <div
                class="cellHeaderContent"
              >
                Pickup Location
              </div>
              <div
                class="cellHeaderContent"
              >
                Drop Location
              </div>
              <div
                class="cellHeaderContent"
              >
                Pickup Time
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
            >
              <div
                class="cellContent"
              >
                Location-B
              </div>
              <div
                class="cellContent"
              >
                Location-A
              </div>
              <div
                class="cellContent"
              >
                Tue 18 Jul 2023 9:30 PM
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-sizeMedium requestDetailTableCell css-bsph2u-MuiTableCell-root"
            >
              <div
                class="cellHeaderContent"
              >
                Vendor Name
              </div>
              <div
                class="cellHeaderContent"
              >
                Vendor Contact
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
            >
              <div
                class="cellContent"
              >
                A1 Cab Services
              </div>
              <div
                class="cellContent"
              >
                919005695229
              </div>
            </td>
          </tr>
        </table>
      </div>
    </div>
  </div>
</body>
`;
